internal/condition/eval/eval_test.go

```diff
@@ -78,104 +77,116 @@ condition correct_ip(ip: string) {
 			conditionMet: true,
 			expectedErr:  "",
 		},
-	}
-	for _, test := range tests {
-		t.Run(test.name, func(t *testing.T) {
-			ctx := context.Background()
+		{
+			name:     "multiple_condition_params_all_provided_and_evaluates_true",
+			tupleKey: tuple.NewTupleKeyWithCondition("document:1", "viewer", "user:maria", "condXY", nil),
+			model: parser.MustTransformDSLToProto(`model
+	schema 1.1
+type user
 
-			ts, err := typesystem.NewAndValidate(context.Background(), test.model)
-			require.NoError(t, err)
+type document
+  relations
+    define can_view: [user with condXY]
 
-			contextStruct, err := structpb.NewStruct(test.context)
-			require.NoError(t, err)
+condition condXY(x: bool, y: bool) {
+	x || y
+}`),
+			context:      map[string]interface{}{"x": true, "y": true},
+			conditionMet: true,
+			expectedErr:  "",
+		},
+		{
+			name:     "multiple_condition_params_none_provided_and_evaluates_false",
+			tupleKey: tuple.NewTupleKeyWithCondition("document:1", "viewer", "user:maria", "condXY", nil),
+			model: parser.MustTransformDSLToProto(`model
+	schema 1.1
+type user
 
-			condEvalResult, err := EvaluateTupleCondition(ctx, test.tupleKey, ts, contextStruct)
-			if err != nil {
-				var evalError *condition.EvaluationError
-				require.ErrorAs(t, err, &evalError)
-				require.EqualError(t, evalError, test.expectedErr)
-			} else {
-				require.Empty(t, test.expectedErr)
-				require.Equal(t, test.conditionMet, condEvalResult.ConditionMet)
-			}
-		})
-	}
-}
+type document
+  relations
+    define can_view: [user with condXY]
 
-// TestDefaultCELEvaluationCost is used to ensure we don't decreasee the default evaluation cost
-// of CEL expressions, which would break API compatibility.
-//
-// Critical paths involving ABAC Condition evaluations use the EvaluateTupleCondition function,
-// and so we test that directly to give us higher confidence we're not introducing a compatibility
-// issue.
-func TestDefaultCELEvaluationCost(t *testing.T) {
-	tests := []struct {
-		name     string
-		model    *openfgav1.AuthorizationModel
-		tupleKey *openfgav1.TupleKey
-		context  map[string]any
-		result   *condition.EvaluationResult
-	}{
+condition condXY(x: bool, y: bool) {
+	x || y
+}`),
+			context:      map[string]interface{}{},
+			conditionMet: false,
+			expectedErr:  "'condXY' - tuple 'document:1#viewer@user:maria' is missing context parameters '[x y]'",
```

Comment:
This test is non-deterministic due to the ordering of the missing params `x` and `y`. If this PR is approved in principle, I will alter the tests to make this deterministic.